autoload/sj/ruby.vim

```diff
@@ -358,14 +358,20 @@ function! sj#ruby#SplitBlock()
   if search('\S\%#', 'Wbn')
     let multiline_block = ' '.multiline_block
   endif
-
-  let body = join(split(body, '\s*;\s*'), "\n")
+  
   let replacement = substitute(body, '^'.pattern.'$', multiline_block, '')
+
   " remove leftover whitespace
   let replacement = substitute(replacement, '\s*\n', '\n', 'g')
 
   call sj#ReplaceMotion('Va{', replacement)
 
+  call search(replacement, 'cW')
+  normal! j0
```

Comment:
@AndrewRadev 
I implemented the second method you described. 
But since search(replacement, 'cW') jumps one line above the body of the "do; end" block, I had to
use "normal! j0" to move to the beginning of the correct line.
It seems to me that moving 1 line lower is kind of dirty, but I'm not sure.

I could use sj#GetMotion('Vi{') but I suspect there might be a cleaner way.
Do you have any suggestions :)?

Reply:
No problem in `j0`, you'll see me do this kind of thing elsewhere as well :). Nothing dirty about using the basic motions.

A practical problem, though, is searching for the replacement. What if the replacement has special characters, like `*`? You can try it with `foo * bar`, which as a pattern is "foo, followed by spaces, followed by bar", which won't actually match the string :). It's possible to search for a literal string by prefixing it with `\V`, but it feels unnecessary. Why the search? Where are you trying to position the cursor? Wouldn't just `j0` be enough to get you in the right place?

Reply:
Oh, great!
I wasn't sure where ReplaceMotion would leave the cursor.

I've left only j0. Less code, less bugs :)